src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -92,12 +108,27 @@ private void FinalizeDispose()
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        /// <summary>Path to the procfs file that contains the maximum number of inotify instances an individual user may create.</summary>
+        private const string MaxUserInstancesPath = "/proc/sys/fs/inotify/max_user_instances";
+
+        /// <summary>Path to the procfs file that contains the maximum number of inotify watches an individual user may create.</summary>
+        private const string MaxUserWatchesPath = "/proc/sys/fs/inotify/max_user_watches";
+
         /// <summary>
         /// Cancellation for the currently running watch operation.  
         /// This is non-null if an operation has been started and null if stopped.
         /// </summary>
         private CancellationTokenSource _cancellation;
 
+        /// <summary>Reads the value of a max user limit path from procfs.</summary>
+        /// <param name="path">The path to read.</param>
+        /// <returns>The value read, or "0" if a failure occurred.</returns>
+        private static string ReadMaxUserLimit(string path)
+        {
+            try { return File.ReadAllText(path).Trim(); } catch { }
+            return "0";
```

Comment:
Since this returns a string anyway, why not return something like "Couldn't read max user limit" or "N/A" to avoid confusion around the value not really being configured to 0 on the system?